### PR TITLE
Adding JSON encoding option: Use proto field names instead of lowerCamelCase names

### PR DESCRIPTION
**Pull request by trad55**
https://developers.google.com/protocol-buffers/docs/proto3#json_options

### CODE REVIEW

File: Sources/SwiftProtobuf/JSONEncodingOptions.swift [hunk 1]
```diff
@@ -18,5 +18,8 @@ public struct JSONEncodingOptions {
   /// Always print enums as ints. By default they are printed as strings.
   public var alwaysPrintEnumsAsInts: Bool = false
 
+  /// Always print proto field name instead of JSON(lowerCamelCase) name. By default they are printed as JSON name.
+  public var alwaysPrintProtoFieldNames: Bool = false
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/JSONEncodingOptions.swift]: `alwaysPrint` doesn't seem right here, as it seems like if you weren't print the field names you'd maybe use the field numbers instead?

The C++ seems to call this [`preserve_proto_field_names`](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/util/json_util.h#L81) ([CHANGES.txt entry](preserve_proto_field_names))

So maybe some naming more along those lines so it better indicates it is picking between the two string forms?

File: Tests/SwiftProtobufTests/Test_JSONEncodingOptions.swift [hunk 2]
```diff
@@ -101,4 +101,86 @@ class Test_JSONEncodingOptions: XCTestCase {
 
   }
 
+  func testAlwaysPrintProtoFieldNames() {
+    var jsonNames = JSONEncodingOptions()
+    jsonNames.alwaysPrintProtoFieldNames = false
+    jsonNames.alwaysPrintEnumsAsInts = true
```
Comment (thomasvl) [inline Tests/SwiftProtobufTests/Test_JSONEncodingOptions.swift]: Why the enums off the default also?

**Comment by trad55** [comment]
@thomasvl Thank you for your comments.
I fixed problems and resolved conflicts by rebasing.

**Comment by thomasvl** [comment]
@tbkka lgtm, naming ok with you?

**Comment by thomasvl** [comment]
Thanks!

**Pull request merged**